src/modules/ModuleStaking/composable.pools.ts

```diff
@@ -12,7 +12,31 @@ export function useMappedPools(props: {
   pools: Ref<undefined | null | PoolsQueryResult>
   rewards: Ref<undefined | null | Rewards<Address>>
   tokens: Ref<undefined | null | Tokens>
+  blockNumber: Ref<number | null>
 }) {
+  const sortedPoolEndBlocks = computed(() => {
+    const {
+      pools: { value: poolsResult },
+    } = props
+    if (!poolsResult) return null
+
+    return poolsResult.pools.map((pool) => Number(pool.endBlock)).sort((a, b) => a - b)
+  })
+
+  // Needed to avoid unnecessary recalculations in main computed function
+  const roundedBlockNumber = computed(() => {
```

Comment:
Maybe even `computedEager`, because the computation is cheap while effects are not.

Reply:
Done